src/components/Ref/Ref.tsx

```diff
@@ -30,15 +41,39 @@ export default class Ref extends React.Component<RefProps> {
     innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
   }
 
+  static getDerivedStateFromProps(props: RefProps) {
+    const child = React.Children.only(props.children)
+
+    return {
+      child,
+      isForward: isForwardRef(child),
+    }
+  }
+
   componentDidMount() {
-    handleRef(this.props.innerRef, findDOMNode(this))
+    if (!this.state.isForward) {
```

Comment:
I am afraid the conditional login here can surprise user of the component. I would rather prefer not to be smart here and introduce two separate `Ref` components - one for forward ref, the other one with `findDOMNode`.